auth/auth.go

```diff
@@ -0,0 +1,202 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"time"
+
+	"google.golang.org/grpc/credentials"
+)
+
+func New(authURL string, insecure bool) (client *Client, err error) {
+	client = &Client{
+		insecure: insecure,
+		api: &http.Client{
+			Transport:     nil,
+			CheckRedirect: nil,
+			Timeout:       30 * time.Second,
+		},
+	}
+
+	if client.endpoint, err = url.Parse(authURL); err != nil {
+		return nil, fmt.Errorf("could not parse auth url: %w", err)
+	}
+
+	if client.api.Jar, err = cookiejar.New(nil); err != nil {
+		return nil, fmt.Errorf("could not create cookiejar: %w", err)
+	}
+
+	return client, nil
+}
+
+type Client struct {
+	endpoint *url.URL
+	api      *http.Client
+	apikey   *APIKey
+	tokens   *Tokens
+	insecure bool
+}
```

Comment:
Creates a new Quarterdeck HTTP client that only does `Authenticate` and `Refresh` to get access tokens to authenticate to Ensign with. 

Note, SC-14982 talks about adding better docstrings and documentation to this package.